src/Hazelcast.Net/Networking/NetworkingOptions.cs

```diff
@@ -130,8 +119,17 @@ private NetworkingOptions(NetworkingOptions other)
         /// <summary>
         /// Gets the connection <see cref="RetryOptions"/>.
         /// </summary>
+        /// <remarks>
+        /// <para>Specifies the Hazelcast client connection parameters, including the timeout, i.e. the maximum
+        /// amount of time that the Hazelcast client can spend trying to establish a connection to the cluster
+        /// before failing. See <see cref="SocketOptions"/> for specifying the individual socket parameters,
+        /// including the individual socket connection timeout.</para>
+        /// </remarks>
         public RetryOptions ConnectionRetry { get; } = new RetryOptions();
 
+        // FIXME document
+        public int TimeoutMilliseconds { get; set; } = 30_000;
```

Comment:
This relates to only connection-related timeout (although shared by others, all should be connection timeout)
so a `ConnectionTimeoutMilliseconds ` can be a better name.

Other clients use 5_000 ms as the default value.

Reply:
oops forgot the FIXME, going to document now - and use proper name & default value